src/arduino-timer.h

```diff
@@ -131,10 +131,14 @@ class Timer {
                 const unsigned long duration = t - task->start;
 
                 if (duration >= task->expires) {
-                    task->repeat = task->handler(task->opaque) && task->repeat;
-
-                    if (task->repeat) task->start = t;
-                    else remove(task);
+                    Task activeId = task_id(task);
+                    bool repeating = task->repeat;
+                    repeating &= task->handler(task->opaque);
+
+                    if (activeId == task_id(task)) {
```

Comment:
If this task was deleted (early), the timer should not delete a new task using the slot.